### PR TITLE
feat(ingest):looker - reduce mem usage, misc reporting improvements

### PR DESCRIPTION
**Pull request by shirshanka**
Main improvements:
- Memory usage reduced significantly
- All api access funneled through a single class
- More streamlined incremental progress and reporting

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 1]
```diff
@@ -1,16 +1,17 @@
+from __future__ import print_function
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:1]: nit - don't need this line

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_lib_wrapper.py [hunk 2]
```diff
@@ -75,10 +87,16 @@ def __init__(self, config: LookerAPIConfig) -> None:
                 "Failed to initialize Looker client. Please check your configuration."
             ) from e
 
-    def get_client(self) -> Looker31SDK:
-        return self.client
+        self.client_stats = LookerAPIStats()
+
+    @staticmethod
+    def __fields_mapper(fields: Union[str, List[str]]) -> str:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_lib_wrapper.py:93]: nit - double leading underscore has a special meaning in python; a single underscore would be stylistically correct

**Review by hsheth2 — APPROVED** [review]
left two minor nits - can be fixed in a follow-up PR

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_lib_wrapper.py (hunk 2 above)
Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_lib_wrapper.py:93]: ack

**Pull request merged**